#! /bin/sh -e
## 1_newpatch.dpatch by Roland Mas <lolando@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: home-end.el: Xemacs compatibility

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
       -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
       -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
	*)
		echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
		exit 1;;
esac

exit 0
				    
diff -urNad 1.tmp/elisp/home-end.el 1/elisp/home-end.el
--- 1.tmp/elisp//emacs-goodies-el/home-end.el	2003-01-05 17:10:41.000000000 +0100
+++ 1/elisp//emacs-goodies-el/home-end.el	2003-01-05 17:21:41.000000000 +0100
@@ -50,8 +50,8 @@
            (key1 (if (> len 0) (elt keys (- len 1)) nil))
            (key2 (if (> len 1) (elt keys (- len 2)) nil))
            (key3 (if (> len 2) (elt keys (- len 3)) nil))
-           (key-equal-1 (eq key1 key2))
-           (key-equal-2 (and key-equal-1 (eq key2 key3))))
+           (key-equal-1 (equal key1 key2))
+           (key-equal-2 (and key-equal-1 (equal key2 key3))))
       (cond (key-equal-2 (goto-char (point-min)))
             (key-equal-1 (push-mark home-end-marker)
                          (move-to-window-line 0))
@@ -71,8 +71,8 @@
            (key1 (if (> len 0) (elt keys (- len 1)) nil))
            (key2 (if (> len 1) (elt keys (- len 2)) nil))
            (key3 (if (> len 2) (elt keys (- len 3)) nil))
-           (key-equal-1 (eq key1 key2))
-           (key-equal-2 (and key-equal-1 (eq key2 key3))))
+           (key-equal-1 (equal key1 key2))
+           (key-equal-2 (and key-equal-1 (equal key2 key3))))
       (cond (key-equal-2 (goto-char (point-max)))
             (key-equal-1 (push-mark home-end-marker)
                          (move-to-window-line -1)
